src/hooks/useStepper.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { ReactElement, ReactNode, useState } from 'react';
+import { StepType } from '@/pages/Payments/Payments';
+
+interface StepProps<T> {
+  name: T;
+  children: ReactNode;
+}
+
+interface StepperProps<T> {
+  children: Array<ReactElement<StepProps<T>>>;
+}
+
+export const useStepper = <T extends StepType>(steps: T) => {
```

Comment:
```suggestion
export const useStepper = <T extends readonly string[]>(steps: T) => {
```

제네릭으로 StepType을 받고있는 상황에서 Payments에 의존적인 정의된 StepType을 해당 훅 내부 모듈에서 직접 사용할 필요는 없을 것 같아요!

공통적으로 사용할 수 있는 훅은 최대한 도메인을 알 수 없도록 하는 편이 유지관리에 유리해요!

Reply:
공통 훅의 경우 의존성을 최대한 낮추는 방향으로 설계함과 동시에 도메인은 최대한 알 수 없도록 하는 것이 유지 관리에 유리하다! 잊지 않겠습니다 ㅎㅎ
추가로 `readonly` 로 제약을 걸어주면 코드의 예측성과 안정성도 높아지겠네요!